near-sdk/src/json_types/mod.rs

```diff
@@ -2,10 +2,26 @@
 
 mod hash;
 mod integers;
-mod public_key;
 mod vector;
 
 pub use hash::Base58CryptoHash;
 pub use integers::{I128, I64, U128, U64};
-pub use public_key::{Base58PublicKey, CurveType};
 pub use vector::Base64VecU8;
+
+// This deprecated attribute doesn't work for the current rust version (1.52)
+// but will likely work in the future. Also included just to note that it is
+// indeed deprecated and not just a random export.
+#[deprecated(
+    since = "4.0.0",
+    note = "This import is deprecated. Best to import directly from near_sdk"
+)]
+pub use crate::types::CurveType;
+use crate::types::PublicKey;
```

Comment:
you might want to move this above the other to not have it grouped or just reference it directly below. Opinionated though as it was just confusing for me to read

Reply:
ahh for sure, I'll move it out of the way